### PR TITLE
Add unparam, golint, gosec, scopelint, govet, unconvert and maligned linters

### PR DESCRIPTION
**Pull request by saschagrunert**
This PR adds the mentioned linters and their fixes within separate commits.

### CODE REVIEW

File: lib/config.go [hunk 1]
```diff
@@ -136,28 +136,38 @@ type RootConfig struct {
...
+	// The format of each mount is "host-path:container-path"
+	DefaultMounts []string `toml:"default_mounts"`
+
+	// Capabilities to add to all containers.
+	DefaultCapabilities []string `toml:"default_capabilities"`
+
+	// Sysctls to add to all containers.
+	DefaultSysctls []string `toml:"default_sysctls"`
+
+	// DefeultUlimits specifies the default ulimits to apply to containers
```
Comment (giuseppe) [inline lib/config.go]: typo in DefeultUlimits

Comment (saschagrunert) [reply lib/config.go]: Thanks for the hint, fixed it.

**Pull request merged**